distribution/lib/Standard/Table/0.2.32-SNAPSHOT/src/Data/Matching.enso

```diff
@@ -0,0 +1,165 @@
+   type Exact (case_sensitivity : (True | Case_Insensitive) = True)
+
+   ## UNSTABLE
+      Regex-based name matching.
+
+      A name is matched if its name matches the provided regular expression.
+   type Regex (case_sensitivity : (True | Case_Insensitive) = True)
+
+## UNSTABLE
+   Specifies that the operation should ignore case.
+
+   TODO: Since case-sensitive can be locale dependent, in the future this may
+         be extended with a `locale` setting.
+type Case_Insensitive
+
+
+## UNSTABLE
+   An error indicating that a criterion did not match any names in the input.
+type No_Matches_Found (criterion : Text)
+
+No_Matches_Found.to_display_text : Text
+No_Matches_Found.to_display_text =
+    "The criterion ["+this.criterion+"] did not match any names in the input."
+
+
+## UNSTABLE
+   Selects objects from an input list that match any of the provided criteria.
+
+   Arguments:
+   - objects: A list of objects to be matched.
+   - criteria: A list of texts representing the matching criteria. Their meaning
+     depends on the matching strategy.
+   - reorder: Specifies whether to reorder the matched objects according to the
+     order of the matching criteria.
+     If `False`, the matched entries are returned in the same order as in the input.
+     If `True`, the matched entries are returned in the order of the criteria
+     matching them. If a single object has been matched by multiple criteria, it
+     is placed in the group belonging to the first matching criterion on the list.
+     If a single criterion's group has more than one element, their relative
+     order is the same as in the input.
+   - name_mapper: A function mapping a provided object to its name, which will
+     then be matched with the criteria. It is set to the identity function by
+     default, thus allowing the input to be a list of names to match. But it can be
+     overridden to enable matching more complex objects.
+   - matching_strategy: A `Matching_Strategy` instance specifying how to
+     interpret the criterion.
+   - on_problems: Specifies the behavior when a problem occurs during the function.
+     By default, a warning is issued, but the operation proceeds.
+     If set to `Report_Error`, the operation fails with a dataflow error.
+     If set to `Ignore`, the operation proceeds without errors or warnings.
+   - warning_callback: A callback specifying the function to call when issuing a
+     warning. This is a temporary workaround to allow for testing the warning
+     mechanism. Once the proper warning system is implemented, this argument
+     will become obsolete and will be removed. No user code should use this
+     argument, as it will be removed in the future.
+
+   > Example
+     Selects objects matching one of the provided patterns, preserving the input order.
+
+         Matching.match_criteria ["foo", "foobar", "quux", "baz", "Foo"] [".*ba.*", "f.*"] matching_strategy=(Regex case_sensitivity=True) == ["foo", "foobar", "baz"]
```

Comment:
It may make sense to allow setting `matching_strategy` per criterion? I'm very unclear about the purpose of this code, so I may be wrong here. Anyway, what I mean is something like the `by` argument in `Table.sort`.

Reply:
This is aligned with the design for Select and other tools, as discussed in https://github.com/enso-org/design/pull/15 so the actual place for changing that granularity would be there but I think it was already mentioned and the decision was to do it as seen here (may remember wrongly, will verify).